plans/network/pingpong.go

```diff
@@ -43,23 +51,23 @@ func pingpong(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 	runenv.RecordMessage("before netclient.MustConfigureNetwork")
 	netclient.MustConfigureNetwork(ctx, config)
 
-	seq := client.MustSignalAndWait(ctx, "ip-allocation", runenv.TestInstanceCount)
-
 	// Make sure that the IP addresses don't change unless we request it.
 	if newAddrs, err := net.InterfaceAddrs(); err != nil {
 		return err
 	} else if !sameAddrs(oldAddrs, newAddrs) {
 		return fmt.Errorf("interfaces changed")
 	}
 
+	seq := client.MustSignalAndWait(ctx, "ip-allocation", runenv.TestInstanceCount)
+
 	runenv.RecordMessage("I am %d", seq)
 
 	ipC := byte((seq >> 8) + 1)
 	ipD := byte(seq)
 
 	config.IPv4 = runenv.TestSubnet
 	config.IPv4.IP = append(config.IPv4.IP[0:2:2], ipC, ipD)
-	config.IPv4.Mask = []byte{255, 255, 255, 0}
+	config.IPv4.Mask = []byte{255, 254, 0, 0}
```

Comment:
Why are we using this mask? Not saying that there's anything wrong with that - I just genuinely don't know.

Reply:
The previous mask (/24) was too restrictive, and instances could not connect to each other over their assigned IP when running the test on EKS. Since the test subnet has a mask of /12 (e.g. 10.32.0.0/12), instances could receive an IP that would be excluded by the /24 mask, for example 10.33.0.0.

If we reduce the mask too much (e.g. /8), then the test will fail locally, as well.
By reducing the mask to the current value (/15), the test will work both locally and on the new EKS cluster.

Reply:
Would you mind adding a comment about this here? I feel like this is a sort of information that could be easily forgotten.

Reply:
Bumping this, sorry if this is silly, I'm catching up with this networking wizardry:

Our EKS Test Subnet might be something like `10.32.0.0/12`.
When we called this code before, it was possible that an instance received the address `10.33.0.1`. While it expected something with the subnet mask `/24`, which means:

- the parameter above `config.IPv4.IP = append(config.IPv4.IP[0:2:2], ipC, ipD)` is not really used in the case of EKS, correct? We get a different IP.
- somehow, that would break this instance results because having to reach an address outside its subnet somehow "broke" the instance, correct?

Now we're going to use a mask `/15`, which means:

- the instance might get assigned the address `10.40.0.01`, which is within the network defined by `10.32.0.0/12` but outside `10.32.0.0/15`.

Is this correct? Am I missing something?
Why do we need a subnet different from the one passed in `TestSubnet`?

Reply:
@brdji This hasn't been resolved yet 🚨

Reply:
Sorry - I missed the comment before merging;

>somehow, that would break this instance results because having to reach an address outside its subnet somehow "broke" the instance, correct?

Correct  - the mask should match the data subnet range, otherwise such an issue can pop up.

>Now we're going to use a mask /15

We've noticed this discrepancy in the other PR - the mask is set to `/12`.

Reply:
You're answering all my questions,

We're going to use a mask `/15`, which means:

- the instance might get assigned the address `10.40.0.01`, which is within the network defined by `10.32.0.0/12` but outside `10.32.0.0/15`.

Is this correct? Am I missing something?


> We don't need a different subnet, and they should actually match up. We've noticed this discrepancy in the other PR and fixed it (for EKS) - the mask is set to /12. Note the comment here: [Pingpong.go](https://github.com/testground/testground/blob/5611a78f4cdfb91026321bea0dfca47e37c08ec3/plans/network/pingpong.go).

What stops us from using the env value, passed by testground, instead of hardcoding it here?

Reply:
> Is this correct? Am I missing something?

You are correct; that's why I mentioned the fix for the EKS PR. We noticed this discrepancy later (after I closed this PR, before responding).

>What stops us from using the env value, passed by testground, instead of hardcoding it here?

Nothing is stopping us, except that the env value must be present (and I'm not certain its present in every runner). Do you wish for me to change the test so that it uses the env value, if possible?

Reply:
Thanks for fixing this in the next PR,

> Nothing is stopping us, except that the env value must be present (and I'm not certain its present in every runner). Do you wish for me to change the test so that it uses the env value, if possible?

That looks like a better option but let's not introduce changes without being certain they are working,
I created a follow-up issue, https://github.com/testground/testground/issues/1486, and added a few notes here. No need to work on this now.